pkg/controller/metrics/metricsWithDynamicLabels.go

```diff
@@ -0,0 +1,91 @@
+package metrics
+
+import (
+	"sort"
+
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	"github.com/openshift/hive/apis/hive/v1/metricsconfig"
+)
+
+// metricsWithDyanamicLabels encompasses metrics that can have optional labels.
+type metricsWithDyanamicLabels interface {
+	LogMetricWithDynamicLabels(metav1.Object, log.FieldLogger, float64)
+}
+
+type CounterWithDynamicLabels struct {
+	*prometheus.CounterVec
+	FixedLabels    map[string]string
+	OptionalLabels map[string]string
+}
+
+// LogMetricWithDynamicLabels logs the counter metric. It simply increments the value, so last parameter isn't used.
+func (c CounterWithDynamicLabels) LogMetricWithDynamicLabels(obj metav1.Object, log log.FieldLogger, _ float64) {
+	curriedVec, err := c.CounterVec.CurryWith(c.FixedLabels)
```

Comment:
As noted above, we can't use whatever's in the global `FixedLabels` here -- this func needs to be invoked with the proper label values either already curried into `c` or via a param that we then curry in.